### PR TITLE
python311Packages.scipy: 1.11.4 -> 1.12.0

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Closes #277306

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
Runnning now:

```
nix build -Lf. --max-jobs 1 --cores 3 {,pkgsi686Linux,pkgsCross.{aarch64,armv7l-hf}-multiplatform}.python310Packages.scipy
```

File: pkgs/development/python-modules/scipy/default.nix [hunk 1]
```diff
@@ -75,13 +78,11 @@ in buildPythonPackage {
...
-  # Relax deps a bit
+  # Upstream complicated numpy version pinning is causing issues in the
+  # configurePhase, so we pass on it.
   postPatch = ''
     substituteInPlace pyproject.toml \
-      --replace 'meson-python>=0.12.1,<0.15.0' 'meson-python' \
-      --replace 'numpy==' 'numpy>=' \
-      --replace "pybind11>=2.10.4,<2.11.1" "pybind11>=2.10.4,<2.12.0" \
-      --replace 'wheel<0.41.0' 'wheel'
+      --replace-fail 'numpy==' 'numpy>=' \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scipy/default.nix:85]: Can we use pythonRelaxDepsHook here?

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by vcunat** [comment]
It fixes scipy build and doesn't look bad, so let's move on.

**Comment by vcunat** [comment]
@DavHau: `python3Packages.scikit-learn` needs to react to this somehow I guess?
https://hydra.nixos.org/build/247211840/nixlog/1/tail

**Comment by doronbehar** [comment]
I can't take responsibility to all consequences of such an update :) Feel free to file a PR for scikit-learn and ping me for a review if you wish.

**Comment by vcunat** [comment]
It's still failing the same for me.

**Comment by bryango** [comment]
@DavHau @doronbehar @vcunat `scikit-learn` failure may be fixed by a version bump, see #283666.

**Pull request merged**